MPU/src/driverio.cpp

```diff
@@ -6,102 +6,76 @@ DRIVERIO::DRIVERIO(){}
 
 DRIVERIO::DRIVERIO(CASCADIAMC *p_motorController, ORIONBMS *p_bms)
 {
-    pinMode(REVERSE_SW_PIN, INPUT_PULLUP);
-
-    pinMode(SS_BUTT_PIN, INPUT_PULLUP);
-    pinMode(SS_LED_PIN, OUTPUT);
-    digitalWrite(SS_LED_PIN, LOW);
-
-    pinMode(LED4_PIN, OUTPUT);
-    digitalWrite(LED4_PIN, LOW);
-
-    pinMode(LED5_PIN, OUTPUT);
-    digitalWrite(LED5_PIN, LOW);
-
-    pinMode(YLED_PIN, OUTPUT);
-    digitalWrite(YLED_PIN, LOW);
-
-    pinMode(SPEAKER_PIN, OUTPUT);
-    digitalWrite(SPEAKER_PIN, HIGH);
+    reverseSwitch = SWITCH(REVERSE_SW_PIN);
+    ssButton = STARTBUTTON(SS_LED_PIN, SS_BUTT_PIN);
+    socLED = LED(LED4_PIN);
+    tempLED = LED(LED5_PIN);
+    yLED = LED(YLED_PIN);
+    speaker = SPEAKER(SPEAKER_PIN);
 
     powerToggle_wait.cancelTimer();
-    ssButton_debounce.cancelTimer();
-    speaker_wait.cancelTimer();
-    tempWarningBlink_wait.cancelTimer();
 
     motorController = p_motorController;
     bms = p_bms;
+
+    //Initializes the motor direction based on the initial switch state
+    if(reverseSwitch.getSwitchState()) motorController->toggleDirection();
 }
 
 
 DRIVERIO::~DRIVERIO(){}
 
 
-void DRIVERIO::handleSSButton()
+void DRIVERIO::handleSSButtonPress()
 {
-    //if the button is still being held during and after the timer runs out, then toggle power
-    if(!ssButton_debounce.isTimerExpired())
-    {
-        if(!digitalRead(SS_BUTT_PIN)) // If released
-        {
-            ssButton_debounce.cancelTimer();
-        }
-        return;
-    }
-    if(ssButtonDebounced() && bms->getChargeMode())
+    //Poll Button
+    ssButton.checkButtonPin();
+
+    //Poll Speaker if it is written high
+    speaker.attemptToStopSpeaker();
+
+    //If the button isn't pressed, do nothing
+    if(!ssButton.isButtonPressed_Pulse()) return;
+
+    //If the BMS is charging
+    if(bms->getChargeMode())
```

Comment:
In a way I wonder if you should just do if/else to be clear for future readers. It should all simplify out in compile right?

Reply:
The point of the "pulse" function is to make it so that the user can hold the start button forever but only power on the car once

Reply:
Also doing it this way removes a layer of nesting that makes the code less readable, where if I had an if/else statement then it would get a bit messier, actually

Think of that return statement as the else statement, just flipped so there is 1 less layer of nesting, and I commented that if the button isn't pressed, then do nothing